homeassistant/components/opower/__init__.py

```diff
@@ -2,16 +2,34 @@
 
 from __future__ import annotations
 
+from opower import select_utility
+
 from homeassistant.const import Platform
 from homeassistant.core import HomeAssistant
+from homeassistant.helpers import issue_registry as ir
 
+from .const import CONF_UTILITY, DOMAIN
 from .coordinator import OpowerConfigEntry, OpowerCoordinator
 
 PLATFORMS: list[Platform] = [Platform.SENSOR]
 
 
 async def async_setup_entry(hass: HomeAssistant, entry: OpowerConfigEntry) -> bool:
     """Set up Opower from a config entry."""
+    utility_name = entry.data[CONF_UTILITY]
+    try:
+        select_utility(utility_name)
+    except ValueError:
+        ir.async_create_issue(
+            hass,
+            DOMAIN,
+            f"unsupported_utility_{entry.entry_id}",
+            is_fixable=False,
+            severity=ir.IssueSeverity.ERROR,
+            translation_key="unsupported_utility",
+            translation_placeholders={"utility": utility_name},
+        )
+        return False
```

Comment:
Why don't we make it a proper repair and propose the user to delete it? Then we immediately make it actionable and easy for the user to fix.

In theory you can copy the `repairs.py` from the `homeassistant` component